plugins/downloader/utils.js

```diff
@@ -3,3 +3,29 @@ const electron = require("electron");
 module.exports.getFolder = (customFolder) =>
 	customFolder || (electron.app || electron.remote.app).getPath("downloads");
 module.exports.defaultMenuDownloadLabel = "Download playlist";
+
+const orderedQualityList = ["maxresdefault", "hqdefault", "mqdefault", "sdddefault"];
+module.exports.UrlToJPG = (imgUrl, videoId) => {
+	if (!imgUrl || imgUrl.includes(".jpg")) return imgUrl;
+	//it will almost never get further than hqdefault
+	for (const quality of orderedQualityList) {
+		if (imgUrl.includes(quality)) {
+			return `https://img.youtube.com/vi/${videoId}/${quality}.jpg`;
+		}
+	}
+	return `https://img.youtube.com/vi/${videoId}/default.jpg`;
+}
```

Comment:
this function will 99.9% of the time:
return a `maxresdefault` webp converted to jpg (quite frequent with newer videos)
or return the original url (is jpg) 

0.1% chance (might even be 0%)
to do more than 1 loop iteration (will only happen if original doesn't contains `.jpg` or `maxresdefault`)

This was mostly coded this way to 100% guarantee a jpg url output